packages/neuron-ui/src/widgets/Calendar/index.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState, useEffect, useMemo } from 'react'
+import { getMonthCalendar, useLocalNames, monthInRange, yearInRange, dateEqual, dayInRange } from './utils'
+import styles from './calendar.module.scss'
+
+interface Option {
+  value: number
+  title: string
+  selectable: boolean
+}
+const Selector = ({ options, onChange }: { options: Option[]; onChange: (option: Option) => void }) => (
+  <ol className={styles.calOptions}>
+    {options.map(option => (
+      <li key={option.value} role="presentation">
+        <button
+          type="button"
+          aria-label={option.title}
+          title={option.title}
+          role="menuitem"
+          onClick={() => onChange(option)}
+          disabled={!option.selectable}
+        >
+          {option.title}
+        </button>
+      </li>
+    ))}
+  </ol>
+)
+
+export interface CalendarProps {
+  value: Date | undefined
+  onChange: (value: Date) => void
+  minDate?: Date
+  maxDate?: Date
+}
+const Calendar: React.FC<CalendarProps> = ({ value, onChange, minDate = null, maxDate = null }) => {
+  const [year, setYear] = useState(new Date().getFullYear())
+  const [month, setMonth] = useState(new Date().getMonth() + 1)
+  const [status, setStatus] = useState<'year' | 'month' | 'date'>('date')
+
+  useEffect(() => {
+    setYear(value?.getFullYear() ?? new Date().getFullYear())
+    setMonth((value?.getMonth() ?? new Date().getMonth()) + 1)
+  }, [value])
+
+  const locale = useLocalNames()
+  const weeknames = useMemo(() => [...Array(7).keys()].map(_ => locale.dayNamesMin[(_ + locale.firstDayOfWeek) % 7]), [
```

Comment:
First, don't instantiate an array in the format `Array(length)`, I cannot find a reference, but it's not a good practice.

Second, `[...Array(7).keys()]` could be replaced by `Array.from({ length: 7 })`

Last, `_` implies the variable is not used, `i` or `index` should be used in this case.